### PR TITLE
Add groups full

### PR DESCRIPTION
**Pull request by atreidas1**

### CODE REVIEW

File: npm-debug.log [hunk 1]
```diff
@@ -0,0 +1,35 @@
+0 info it worked if it ends with ok
```
Comment (ghost) [inline npm-debug.log]: it shouldn't be under git

File: css/style.css [hunk 2]
```diff
@@ -1,109 +1,95 @@
-.completed{
+.completed {
```
Comment (ghost) [inline css/style.css:1]: do we really need this css file ?

File: app/components/login/login.controller.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        .then(loginSucces, loginError);
+    }
+
+    function loginSucces(response) {
+      SessionUser.getUser();
+      $state.go("groups.list");
+    }
+
+    function loginError(response) {
+      var responseStatus = response.status;
```
Comment (ghost) [inline app/components/login/login.controller.js]: I think var `responseStatus` is not used anywhere

File: app/components/login/login.controller.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    $scope.signIn = signIn;
+    $scope.isMsgHide = true;
+
+    function signIn() {
+      $http.post(API.BASE + API.AUTH + API.LOGIN, $scope.user)
+        .then(loginSucces, loginError);
+    }
+
+    function loginSucces(response) {
+      SessionUser.getUser();
```
Comment (ghost) [inline app/components/login/login.controller.js]: `getUser` is async, it returns promise. I think it's better to change state after it finished

**Comment by ghost** [comment]
it's a lot of code, hard to review. But I have some comments. Before action merge you can squash your comments to the one - http://davidwalsh.name/squash-commits-git

File: app/app.resource.conf.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+;
+(function () {
+  angular
+    .module("todo")
+    .config(resourceConfig);
+
+  function resourceConfig($resourceProvider) {
```
Comment (NikitaSedyx) [inline app/app.resource.conf.js]: move it to app/app.config.js, plz

File: app/app.route.js [hunk 6]
```diff
@@ -1,37 +1,47 @@
-;(function(){
+;(function () {
   angular
     .module("todo")
 
-    .config(RouterConfig)
+  .config(RouterConfig)
```
Comment (NikitaSedyx) [inline app/app.route.js:5]: keep spaces here

File: app/components/groups/add-groups.jade [hunk 7]
```diff
@@ -0,0 +1,78 @@
+.row.new-groups-inp(
+  ng-controller="AddGroupController"
+  ng-switch="actions.action"
+)
+  .col-md-6.col-md-offset-3.new-group(
+    ng-switch-when="default"
+    ng-click="actions.action='new-task'"
```
Comment (NikitaSedyx) [inline app/components/groups/add-groups.jade]: mb name new-group is better

File: app/components/groups/add-groups.jade [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        tooltip="Create new list"
+        ng-click="actions.action='new-task';listMode(view)"
+      )
+      i.glyphicon.glyphicon-search.ctrl-btn(
+        tooltip-placement="bottom"
+        tooltip="Search"
+        ng-click="actions.action='search'"
+      )
+  .col-md-6.col-md-offset-3.new-group(ng-switch-when="new-task")
+    textarea.task-title(
```
Comment (NikitaSedyx) [inline app/components/groups/add-groups.jade]: class name should be `group-title` here

File: app/components/groups/group-list/group-list.controller.js [hunk 9]
```diff
@@ -1,22 +1,23 @@
-;(function(){
+;
+(function () {
   angular
     .module("todo")
 
-    .controller("GroupListController", GroupListController)
+  .controller("GroupListController", GroupListController)
```
Comment (NikitaSedyx) [inline app/components/groups/group-list/group-list.controller.js:6]: the same, keep spaces

File: app/components/groups/group-storage.service.js [hunk 10]
```diff
@@ -1,42 +1,45 @@
...
-        data: [],
-        totalItems: 0
-      }
-      self.loadData = loadData
-      
-      var loaders = {
-        "list": listLoader,
-        "panel": panelLoader
-      }
+    .service("GroupStorage", GroupStorage)
```
Comment (NikitaSedyx) [inline app/components/groups/group-storage.service.js:6]: the same

File: app/components/groups/group.contoller.js [hunk 11]
```diff
@@ -4,18 +4,20 @@
...
-    function GroupController(GroupStorage, $scope){
+    function GroupController(GroupStorage, $scope, SessionUser){
       $scope.groupView = "list"
       $scope.groups = GroupStorage.groups
       $scope.params = {
         offset: 0
       }
       $scope.changeGroupView = changeGroupView
-      function changeGroupView(){
+      function changeGroupView(groupView){
```
Comment (NikitaSedyx) [inline app/components/groups/group.contoller.js]: we have `$scope.groupView`

File: app/components/groups/group.resource.js [hunk 12]
```diff
@@ -1,25 +1,28 @@
...
-          method: "POST"
-        },
-        editGroup: {
-          method: "PUT"
-        },
-        deleteGroup: {
-          method: "DELETE"
-        }
-      })
+  GroupResource.$inject=["$resource", "API"]
```
Comment (NikitaSedyx) [inline app/components/groups/group.resource.js:5]: the same, keep spaces, plz

File: app/components/groups/groups.jade [hunk 13]
```diff
@@ -1,19 +1,19 @@
-.row
-  //new group will be here
+include ./add-groups.jade
 .row
   .btn-group.switch.pull-right
     label.btn.btn-primary(
       ng-model="groupView"
       btn-radio="'list'"
-      ng-change="changeGroupView()"
+      ng-change="changeGroupView('list')"
```
Comment (NikitaSedyx) [inline app/components/groups/groups.jade]: `ng-model="groupView"`

File: app/components/login/login.controller.js (hunk 4 above)
Comment (NikitaSedyx) [inline app/components/login/login.controller.js]: `getUser` doesn't return promise
for me it's normal, mb name is bad

File: assets/sass/_register-and-login.sass [hunk 14]
```diff
@@ -0,0 +1,17 @@
+.login-form, .register-form
+	margin-top: 120px
```
Comment (NikitaSedyx) [inline assets/sass/_register-and-login.sass:2]: indents

File: app/app.config.js [hunk 15]
```diff
@@ -22,3 +22,13 @@
       })
     })
 })()
+
+;(function () {
```
Comment (NikitaSedyx) [inline app/app.config.js]: for what it? could you insert .config in previous part

File: app/components/groups/add-group.controller.js [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+    function addTask(item) {
+      var newItem = {
+        description: item.description,
+        is_completed: false
+      };
+      $scope.newGroup.items.push(newItem);
+      item.description = undefined;
+    }
+
+    function listMode(view) {
```
Comment (NikitaSedyx) [inline app/components/groups/add-group.controller.js]: `view` isn't used anywhere

File: app/components/groups/add-groups.jade [hunk 17]
```diff
@@ -0,0 +1,78 @@
...
+  .col-md-6.col-md-offset-3.new-group(
+    ng-switch-when="default"
+    ng-click="actions.action='new-task'"
+  )
+    span New group
+    span.ctl-pane.pull-right
+      i.glyphicon.glyphicon-th-list.ctrl-btn(
+        tooltip-placement="bottom"
+        tooltip="Create new list"
+        ng-click="actions.action='new-task';listMode(view)"
```
Comment (NikitaSedyx) [inline app/components/groups/add-groups.jade]: mb function will be better for it

File: app/components/groups/group.contoller.js [hunk 18]
```diff
@@ -4,9 +4,9 @@
 
     .controller("GroupController", GroupController)
 
-    GroupController.$inject = ["GroupStorage", "$scope"]
+    GroupController.$inject = ["GroupStorage", "$scope", "SessionUser"]
 
-    function GroupController(GroupStorage, $scope){
+    function GroupController(GroupStorage, $scope, SessionUser){
       $scope.groupView = "list"
```
Comment (NikitaSedyx) [inline app/components/groups/group.contoller.js:10]: I think `$scope.groupView` should be equal `SessionUser.user.userView`

File: app/components/register/register.controller.js [hunk 19]
```diff
@@ -0,0 +1,38 @@
...
+  angular
+    .module("todo")
+    .controller("RegisterController", RegisterController);
+
+  RegisterController.$inject = ["$scope", "$http", "$state", "API", "SessionUser"];
+
+  function RegisterController($scope, $http, $state, API, SessionUser) {
+    $scope.signUp = signUp;
+    $scope.isMsgHide = true;
+    this.registerSucces = registerSucces;
```
Comment (NikitaSedyx) [inline app/components/register/register.controller.js:12]: it isn't necessary for me

**Pull request closed without merging**